hazelcast/src/test/java/com/hazelcast/internal/partition/MigrationInvocationsSafetyTest.java

```diff
@@ -0,0 +1,152 @@
+        final HazelcastInstance initialMaster = factory.newHazelcastInstance();
+        final HazelcastInstance nextMaster = factory.newHazelcastInstance();
+        final HazelcastInstance slave1 = factory.newHazelcastInstance();
+        final HazelcastInstance slave2 = factory.newHazelcastInstance();
+        final HazelcastInstance slave3 = factory.newHazelcastInstance();
+
+        assertClusterSizeEventually(5, nextMaster, slave1, slave2, slave3);
+
+        // nextMaster & slave1 won't receive partition table updates from initialMaster.
+        dropOperationsBetween(initialMaster, asList(slave1, nextMaster), F_ID, singletonList(PARTITION_STATE_OP));
+
+        warmUpPartitions(initialMaster, slave2, slave3);
+
+        final int initialPartitionStateVersion = getPartitionService(initialMaster).getPartitionStateVersion();
+        assertEquals(initialPartitionStateVersion, getPartitionService(slave2).getPartitionStateVersion());
+        assertEquals(initialPartitionStateVersion, getPartitionService(slave3).getPartitionStateVersion());
+        assertEquals(0, getPartitionService(nextMaster).getPartitionStateVersion());
+        assertEquals(0, getPartitionService(slave1).getPartitionStateVersion());
+
+        dropOperationsBetween(nextMaster, slave3, F_ID, singletonList(FETCH_PARTITION_STATE));
+
+        terminateInstance(initialMaster);
+
+        spawn(new Runnable() {
+            @Override
+            public void run() {
+                assertClusterSizeEventually(4, nextMaster, slave1, slave2, slave3);
+                sleepSeconds(10);
+                resetPacketFiltersFrom(nextMaster);
+            }
+        });
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                int nextPartitionStateVersion = getPartitionService(nextMaster).getPartitionStateVersion();
+                assertThat(nextPartitionStateVersion, greaterThan(initialPartitionStateVersion));
+
+                assertEquals(nextPartitionStateVersion, getPartitionService(slave1).getPartitionStateVersion());
+                assertEquals(nextPartitionStateVersion, getPartitionService(slave2).getPartitionStateVersion());
+                assertEquals(nextPartitionStateVersion, getPartitionService(slave3).getPartitionStateVersion());
+            }
+        });
+
+    }
+
+    @Test
+    public void members_shouldAgree_onPartitionTable_whenMasterChanges_and_anotherMemberCrashes() {
+        final HazelcastInstance initialMaster = factory.newHazelcastInstance();
+        final HazelcastInstance nextMaster = factory.newHazelcastInstance();
+        final HazelcastInstance slave1 = factory.newHazelcastInstance();
+        final HazelcastInstance slave2 = factory.newHazelcastInstance();
+        final HazelcastInstance slave3 = factory.newHazelcastInstance();
+
+        assertClusterSizeEventually(5, nextMaster, slave1, slave2, slave3);
```

Comment:
Any reason that `initialMaster` is not part of the list? Could be replaced with `assertClusterSizeEventually(5, factory.getAllHazelcastInstances())`

Reply:
No reason. If rest of the cluster observes 5 members, then it's guaranteed that the master (`initialMaster` here) observes too.